platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/turtle/runner/Turtle.java

```diff
@@ -132,57 +134,73 @@ public void start() {
+    public void simulateTimeAndValidate(@NonNull final Duration duration) {
         final Instant simulatedStart = time.now();
         final Instant simulatedEnd = simulatedStart.plus(duration);
 
-        Instant nextValidationTime = simulatedStart.plus(validationInterval);
         while (time.now().isBefore(simulatedEnd)) {
             reportThePassageOfTime();
 
             time.tick(simulationGranularity);
             network.tick(time.now());
             tickAllNodes();
-
-            if (time.now().isAfter(nextValidationTime)) {
-                validateConsensusRounds();
-                nextValidationTime = time.now().plus(validationInterval);
-            }
+            validateConsensusRounds();
         }
     }
 
     /**
-     * Validate all collected {@link ConsensusRound} instances during Turtle execution using the suitable validator.
+     * Validate all commonly collected {@link ConsensusRound} instances by all nodes during Turtle execution
+     * using the configured validators.
      *
-     * At the end of the validation, all collected items are cleared to keep memory usage low.
+     * At the end of the validation, the specified commonly collected items are cleared to keep memory usage low.
      */
     public void validateConsensusRounds() {
+        final Set<Long> commonConsensusRounds = getCommonlyCollectedConsensusRounds();
+
         final TurtleNode node1 = nodes.getFirst();
         final List<ConsensusRound> consensusRoundsForNode1 =
-                node1.getConsensusRoundsHolder().getCollectedRounds();
+                node1.getConsensusRoundsHolder().getCollectedRounds().entrySet().stream()
+                        .filter(e -> commonConsensusRounds.contains(e.getKey()))
+                        .map(Map.Entry::getValue)
+                        .collect(Collectors.toList());
+
+        if (!commonConsensusRounds.isEmpty()) {
+            for (int i = 1; i < nodes.size(); i++) {
+                final TurtleNode otherNode = nodes.get(i);
+                final List<ConsensusRound> consensusRoundsForOtherNode =
+                        otherNode.getConsensusRoundsHolder().getCollectedRounds().entrySet().stream()
+                                .filter(e -> commonConsensusRounds.contains(e.getKey()))
+                                .map(Map.Entry::getValue)
+                                .collect(Collectors.toList());
+
+                consensusRoundValidator.validate(consensusRoundsForNode1, consensusRoundsForOtherNode);
+
+                otherNode.getConsensusRoundsHolder().clear(commonConsensusRounds);
+            }
 
+            node1.getConsensusRoundsHolder().clear(commonConsensusRounds);
+        }
+    }
+
+    /**
+     * Collect rounds that reached consensus in all nodes participating in the Turtle network.
+     *
+     * @return the set of round numbers that represent rounds that reached consensus in all nodes
+     */
+    private Set<Long> getCommonlyCollectedConsensusRounds() {
```

Comment:
nit: this method returns round numbers, not rounds, so I suggest this naming simplification
```suggestion
    private Set<Long> getCommonConsensusRoundNums() {
```

Reply:
Renamed